cdap-app-fabric/src/main/java/io/cdap/cdap/internal/state/AppStateStore.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.state;
+
+import io.cdap.cdap.spi.data.InvalidFieldException;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.TableNotFoundException;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.StoreDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * Store for app state data.
+ */
+public class AppStateStore {
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  public AppStateStore(TransactionRunner transactionRunner) {
```

Comment:
We actually don't want the TransactionRunner to be within this class, but outside of it. This is because we will need to be able to perform operations on this table in the same transaction as operations on another table. See the PreferencesTable for a better example of how we want to structure it. Let's rename this class to AppStateTable and remove the transaction stuff from it.

We don't actually follow this pattern in many of our existing classes, which is bad practice.